### ISSUE TITLE
Blacklist issue...

### ISSUE THREAD

**Issue by dwhagar** [bovender/LinkTitles#22]
Ok, so this one is not me, as far as I can tell...  So, first, I had a page called "Turned", now that being a common word, it was getting unintended links.  So, I tried to add it to the Blacklist:

```
$wgLinkTitlesBlackList[] = 'Drone';
$wgLinkTitlesBlackList[] = 'Drones';
$wgLinkTitlesBlackList[] = 'Federation';
$wgLinkTitlesBlacklist[] = 'Turned';
```

Along with my other Blacklisted pages.

This didn't work, every time "Turned" still got auto-linked.  So, I tried to use the keyword:
`__NOAUTOLINKTARGET__`
Was added to the TOP of the page.  Still, no effect the word "turned" or "Turned" kept getting automatically linked.  I finally gave up and changed the page title to "The Turned" since it was a viable replacement.  Still, I can't figure out why it would do this, either of those options should have blocked the page from being autolinked according to the documentation.

Any ideas?

**Comment by bovender** [comment]
Hi @dwhagar,

when you reported this issue, did you have the extension add links on editing a page or on rendering? In other words, might the second part of your post be related to #23?

**Comment by dwhagar** [comment]
This issue is still going on.  I have several pages with the __NOAUTOLINKTARGET__ magic word added to the beginning of the page, but it still gets auto-linked as well as if I add the link name to the blacklist in LocalSettings.php.  I have caching turned on, so I'm going to wait and see about the cache clearing.

**Comment by bovender** [comment]
You can bypass the cache by adding `?action=purge` to the URL. Let me know if this changes anything.

**Comment by bovender** [comment]
@dwhagar I finally found out what might be the problem.

Depending on your setting of `$wgCapitalLinks`, you need to capitalize the first letter of the pages in the black list array. In your example, they are all capitalized, so it really "should" work. I have finally implemented automatic tests for the extension, where there is no issue with the black list if the list is built correctly (i.e., respecting the `$wgCapitalLinks` setting).

Closing this for now. If you are still experiencing this problem, please re-open the issue. We would then have to define the failing scenario in more detail so I can implement a test for it.

**Comment by bovender** [comment]
@dwhagar Scratch what I wrote this morning. Now the tests failed. I fixed something, it will appear in version 5.0.0.
